base_custom_attribute/models/attribute_attribute.py

```diff
@@ -86,25 +82,31 @@ def _build_attribute_field(self, page, attribute):
         return parent
 
     @api.model
-    def _build_attributes_notebook(self, attribute_group_ids):
+    def _build_attributes_notebook(self, attribute_set_id):
```

Comment:
You need to change the logic here.
Now we do not care about the attribute_set_id selected (remove it).
What you are going to do is to build a view with all the attribute you have (that are used by a attribute_set)
Then for each field do something like

<field name="x_my_custom_field" attrs="{'invisible': [('attribute_set_id', 'not in', attribute_set_ids)])>

And the list of attributes set ids are my_custom_field.attribute_set_ids.ids

So every  field will be in the view but only the field used by an attribute set will be display.

Reply:
I tried to do that now, let me know what you think about it!